scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py

```diff
@@ -22,184 +22,321 @@
-    return Conf.get(self._index, 'cluster>server_nodes')[machine_id]
 
-def restart_services(services):
+    check_type(machine_id, str, "machine-id")
+    if len(machine_id) != MACHINE_ID_LEN:
+        raise MotrError(errno.EINVAL, "Invalid machine-id length."
+                        f" Expected: {MACHINE_ID_LEN}"
+                        f" Actual: {len(machine_id)}")
+
+    try:
+        current_node = Conf.get(self._index, 'cluster>server_nodes')[machine_id]
+    except:
+        raise MotrError(errno.EINVAL, "Current node not found")
+
+    check_type(current_node, str, "current node")
+    return current_node
+
+
+def restart_services(self, services):
     for service in services:
-        cmd = "service {} stop".format(service)
-        execute_command(cmd, TIMEOUT_SECS)
-        cmd = "service {} start".format(service)
-        execute_command(cmd, TIMEOUT_SECS)
-        cmd = "service {} status".format(service)
-        execute_command(cmd, TIMEOUT_SECS)
+        sys.stdout.write(f"Restarting {service} service\n")
+        cmd = f"systemctl stop {service}"
+        execute_command(self, cmd)
+        cmd = f"systemctl start {service}"
+        execute_command(self, cmd)
+        cmd = f"systemctl status {service}"
+        execute_command(self, cmd)
 
 def validate_file(file):
     if not os.path.exists(file):
         raise MotrError(errno.ENOENT, "{} not exist".format(file))
 
 def is_hw_node(self):
-    node_type = Conf.get(self._index, f'cluster>{self._server_id}')['node_type']
+    try:
+        node_type = Conf.get(self._index,
+                    f'cluster>{self._server_id}')['node_type']
+    except:
+        raise MotrError(errno.EINVAL, "node_type not found")
+    check_type(node_type, str, "node type")
     if node_type == "HW":
         return True
     else:
         return False
 
 def validate_motr_rpm(self):
-    try:
-        cmd = "uname -r"
-        cmd_res = execute_command(cmd, TIMEOUT_SECS)
-        op = cmd_res[0]
-        kernel_ver = op.replace('\n', '')
-        kernel_module = f"/lib/modules/{kernel_ver}/kernel/fs/motr/m0tr.ko"
-        sys.stdout.write(f"[INFO] Checking for {kernel_module}\n")
-        validate_file(kernel_module)
-        sys.stdout.write(f"[INFO] Checking for {MOTR_SYS_FILE}\n")
-        validate_file(MOTR_SYS_FILE)
-    except MotrError as e:
-        sys.stderr.write("Validate motr rpm failed\n")
-        sys.exit(e._rc)
+    '''
+        1. check m0tr.ko exists in current kernel modules
+        2. check /etc/sysconfig/motr
+    '''
+    cmd = "uname -r"
+    cmd_res = execute_command(self, cmd)
+    op = cmd_res[0]
+    kernel_ver = op.replace('\n', '')
+    check_type(kernel_ver, str, "kernel version")
+
+    kernel_module = f"/lib/modules/{kernel_ver}/kernel/fs/motr/m0tr.ko"
+    sys.stdout.write(f"Checking for {kernel_module}\n")
+    validate_file(kernel_module)
+
+    sys.stdout.write(f"Checking for {MOTR_SYS_FILE}\n")
+    validate_file(MOTR_SYS_FILE)
+
```

Comment:
Why is the "try except" block removed?

It is a best practice to have "try except".

Reply:
```
def main(argv: dict):
    try:	    try:
        desc = "Motr Setup command"	        desc = "Motr Setup command"
        command = Cmd.get_command(desc, argv[1:])	        command = Cmd.get_command(desc, argv[1:])
        command.process()	        command.process()


    except Exception as e:	    except Exception as e:
        sys.stderr.write("error: %s\n\n" % str(e))	        sys.stderr.write(f"\n{str(e)}\n\n")
        sys.stderr.write("%s\n" % traceback.format_exc())	        sys.stderr.write(f"{traceback.format_exc()}\n")
        Cmd.usage(argv[0])	        Cmd.usage(argv[0])
        return errno.EINVAL	        return errno.EINVAL
```
exception caught here in main.